theano/gpuarray/ctc.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import absolute_import, print_function, division
+
+import theano
+from theano import (config, gof)
+import theano.tensor as T
+from .basic_ops import (gpu_contiguous, as_gpuarray_variable, infer_context_name)
+import theano.tensor.nnet.ctc
+from .type import (GpuArrayType, gpu_context_type)
+from .elemwise import GpuDimShuffle
+from theano.gradient import grad_undefined
+from theano.gof import local_optimizer
+from theano.tensor.opt import register_canonicalize
+from theano.tensor.opt import register_stabilize
+
+import os
+import os.path
+from . import pygpu
+
+ctc_enabled = config.ctc.enabled
+
+
+class GpuConnectionistTemporalClassification(gof.COp):
+    """
+    GPU wrapper for Baidu CTC loss function.
+
+    Parameters
+    ----------
+    compute_grad
+        If set to True, enables the computation of gradients of the CTC loss function.
+    """
+    __props__ = ('compute_grad', 'default_output',)
```

Comment:
`default_output` does not need to be in `__props__` list (it is more a class attribute, not an Op property).